src/poliastro/czml/extract_czml.py

```diff
@@ -494,5 +494,158 @@ def add_orbit(
 
         self.i += 1
 
+    def add_trajectory(
+        self,
+        positions,
+        epochs,
+        groundtrack_show=False,
+        groundtrack_lead_time=None,
+        groundtrack_trail_time=None,
+        groundtrack_width=None,
+        groundtrack_color=None,
+        id_name=None,
+        id_description=None,
+        path_width=None,
+        path_show=None,
+        path_color=None,
+        label_fill_color=None,
+        label_outline_color=None,
+        label_font=None,
+        label_text=None,
+        label_show=None,
+    ):
+        """
+        Adds trajectory
+
+        Parameters
+        ----------
+        positions: ~astropy.coordinates.CartesianRepresentation
+            Trajectory to plot.
+        epochs: ~astropy.time.core.Time
+            Epochs for positions.
+        Groundtrack parameters:
+        -----------------------
+
+        groundtrack_show: bool
+            If set to true, the groundtrack is
+            displayed.
+        groundtrack_lead_time: double
+            The time the animation is ahead of the real-time groundtrack
+        groundtrack_trail_time: double
+            The time the animation is behind the real-time groundtrack
+        groundtrack_width: int
+            Groundtrack width
+        groundtrack_color: list (int)
+            Rgba groundtrack color. By default, it is set to the path color
+
+        Id parameters:
+        --------------
+
+        id_name: str
+            Set orbit name
+        id_description: str
+            Set orbit description
+
+        Path parameters
+        ---------------
+
+        path_width: int
+            Path width
+        path_show: bool
+            Indicates whether the path is visible
+        path_color: list (int)
+            Rgba path color
+
+        Label parameters
+        ----------
+
+        label_fill_color: list (int)
+            Fill Color in rgba format
+        label_outline_color: list (int)
+            Outline Color in rgba format
+        label_font: str
+            Set label font style and size (CSS syntax)
+        label_text: str
+            Set label text
+        label_show: bool
+            Indicates whether the label is visible
+
+        """
+        positions = (
+            positions.represent_as(CartesianRepresentation).get_xyz(1).to(u.meter).value
+        )
+        epochs = Time(epochs, format="isot")
+
+        if epochs.__len__() != positions.__len__():
+            raise ValueError("Number of Points and Epochs must be equal.")
+
+        positionList = np.array([])
+
+        for i in range(positions.__len__()):
+            epoch = (epochs[i] - epochs[0]).to(u.second).value
```

Comment:
I wish there was a way to avoid this `for` loop... If we are doing `epochs[i] - epochs[0]` every time, isn't it easier to store `epochs - epochs[0]` somewhere?

Reply:
The ```for``` loop was mainly to append the epochs and positions into a single array. A vectorized utility function should work maybe for epoch calculation?

Reply:
@astrojuanlu I have used a ```lambda``` for epoch calculation and ```zip``` to arrange epochs and positions. How does this sound?